_posts/2016-04-05-extension-framework-game-plan.md

```diff
@@ -0,0 +1,154 @@
+  - **extending Product schema** to   have associations as needed
+  - **support functions in Product Model** too retrieve all users who favorited a product
+
+
+### Request Layer Changes
+
+- Ability to mark and unmark a Product as favorite
+  - a View probably showing all products with ability to mark/unmark product as favorite
+  - a controller action preparing the view
+  - a route exposing the controller / view    through Web
+  - a controller action handling mark product as favorite
+  - a route exposing the controller / view through Web
+  - a controller actionion handling unmark already favorited product
+  - a route exposing the controllerontroller / view through Web
+- Showing All Products liked by a User
+  - a View showing list of Products
+  - a controller preparing the view
+  - already route exposing the controller / view through Web
+- Showing Users who favorited a particular Product
+  - a View showing list of Users
+  - a controller preparing the view
+  - a route exposing the controller / view through Web
+
+_**translates to**_
+
+- Ability to mark and unmark a Product as favorite
+  - a **View** probably showing all products with ability to mark/unmark product as favorite
+  - a **controller** with index / create / delete action
+  - a **route** exposingposing index / create / delete action
+- Showing All Products liked by a User
+  - a **View** showing list of Products
+  - a **controller** preparing the viewew
+  - a **route** exposing the controller / view through Web
+- Showing Users who favorited a particular Product
+  - a **View** showing list of  Users
+  - a **controller** preparing the view
+  - a **route** exposing throughe controller / view through Web
+
+## What we need
+
+- way to extend schema definitions for existing models
+- way to add new functions in existing models
+- way to add routes
+- way to add controller / views for newly added routes
+- way to extend views
+- way to reuse layouts
+- way to reuse already available routes
+
+## How we attempt to solve
+
+- Elixir Metaprogramming
+- Elixir umbrella app dependencies to share and reuse code among Nectar & Extensions using ExtensionManager
+- Extensions as Phoenix project leveraging NectarCommerce
+
+## Bridging the Gap
+
+**Next Posts** would refer the *Favorite Product Extension* to help co-relate and reveal the challenges & solutions implemented to propose an Extension framework.
+
+_Our aim with these posts is to start a dialog with the Elixir community on validity and technical soundness of our approach. We would really appreciate your feedback and reviews, and any ideas/suggestions/pull requests for improvements to our current implementation or entirely different and better way to do things to achieve the goals we have set out for NectarCommerce._
+
+We look forward for your support and feedback on [twitter](https://twitter.com/NectarCommerce) and [github](https://github.com/vinsol/nectarcommerce/pull/48)
+
```

Comment:
As we are asking for support and feedback so thought of adding the link of PR for same
but I still have to replace the link from #48 to #47 :)